pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -196,11 +197,21 @@ func (rc *remoteClient) startWatcher(ctx context.Context, kind string, w multiKu
 	go func() {
 		log.V(2).Info("Starting watch")
 		for r := range newWatcher.ResultChan() {
-			wlKey, err := w.GetWorkloadKey(r.Object)
-			if err != nil {
-				log.V(2).Error(err, "Cannot get workload key", "jobKind", r.Object.GetObjectKind().GroupVersionKind())
-			} else {
-				rc.queueWorkloadEvent(ctx, wlKey)
+			switch r.Type {
+			case watch.Error:
+				switch s := r.Object.(type) {
+				case *metav1.Status:
+					log.V(5).Info("Watch error", "status", s.Status, "message", s.Message, "reason", s.Reason)
+				default:
+					log.V(5).Info("Watch error with unexpected type", "type", fmt.Sprintf("%T", s))
```

Comment:
```suggestion
					log.V(2).Error("Watch error with unexpected type", "type", fmt.Sprintf("%T", s))
```
If this is unexpected, we prefer to get it logged at high level.

Reply:
The type of the error is unexpected, the fact that the error is reported is as expected as two lines above.

Reply:
The message "Watch error with unexpected type" suggests something is going seriously wrong, so setting (5) things going badly. WDYT?

Reply:
Oh, I see you don't have err at hand here to call `Error`, still, v(2) seems more appropriate

Reply:
Let's go with V(3)

Reply:
Let's go with V(3)